### PR TITLE
Use flexbox for table header

### PR DESCRIPTION
**Pull request by tmpacifitech**

### CODE REVIEW

File: components/dash-table/src/dash-table/components/Table/Table.less [hunk 1]
```diff
@@ -504,13 +516,22 @@
...
+                    &--hide,
+                    &--sort {
+                        .not-selectable();
+                        cursor: pointer;
+                    }
+
+                    > *:only-child {
+                        margin-left: auto;
+                    }
                 }
```
Comment (tmpacifitech) [inline components/dash-table/src/dash-table/components/Table/Table.less:535]: This is the only changes on this less file.. other changes are just prettier fix.

**Comment by tmpacifitech** [comment]
Another fix for https://github.com/plotly/dash-table/issues/867

**Review by akksi — CHANGES_REQUESTED** [review]
- Looks like one of the Percy tests is failing (https://app.circleci.com/pipelines/github/plotly/dash/2707/workflows/6936fe57-0c84-4d9f-b4b7-c47452fe9a6e/jobs/44310)
- If there are multiple buttons/icons in a column, they should be wrapped in a container so all of them would be aligned left OR, maybe it would work to always add `margin-left: auto` to the text, so it will put the spacing between the icons and the text (I like the second solution better).
- Do we need to keep the `float: left` properties?

**Comment by alexcjohnson** [comment]
Looks like we had multiple parallel fixes for this... 🙈

**Pull request closed without merging**